cookiecutter/migrate.py

```diff
@@ -22,203 +22,190 @@
-
-            # Build the filterwarnings block
-            new_lines.append(f"{indent}filterwarnings = [\n")
-            # This is fine, indent is defined only once, so even if it is a closure
-            # bound late, the value will always be the same.
-            # pylint: disable-next=cell-var-from-loop
-            new_lines.extend(map(lambda s: f"{indent}  {s}\n", base_specs))
-            for spec in extra_specs:
-                new_lines.append(f'{indent}  "{spec}",\n')
-            new_lines.append(f"{indent}]\n")
-
-            modified = True
-        else:
-            new_lines.append(line)
-
-    if modified and not has_filterwarnings:
-        print(f"Updated {path} to use filterwarnings.")
-        path.write_text("".join(new_lines), encoding="utf-8")
-        return
+def create_dependabot_auto_merge_workflow() -> None:
+    """Create the Dependabot auto-merge workflow file."""
+    workflow_dir = Path(".github") / "workflows"
+    workflow_dir.mkdir(parents=True, exist_ok=True)
+
+    workflow_content = """name: Auto-merge Dependabot PRs
+
+on:
+  pull_request:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-merge:
+    if: github.actor == 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Auto-merge Dependabot PR
+        uses: frequenz-floss/dependabot-auto-approve@3cad5f42e79296505473325ac6636be897c8b8a1 # v1.3.2  # noqa: E501
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          dependency-type: 'all'
+          auto-merge: 'true'
+          merge-method: 'merge'
+          add-label: 'auto-merged'
+"""
+
+    workflow_file = None
+    action_pattern = "dependabot-auto-approve"
+
+    if workflow_dir.exists():
+        for file in workflow_dir.glob("*.y*ml"):
+            content = file.read_text(encoding="utf-8")
+            if action_pattern in content:
+                workflow_file = file
+                print(f"Found existing workflow using {action_pattern}: {file}")
+                break
+
+    if workflow_file is None:
+        workflow_file = workflow_dir / "auto-dependabot.yaml"
+
+    file_exists = workflow_file.exists()
+    if file_exists:
+        existing_content = workflow_file.read_text(encoding="utf-8")
+        # Check for auto-merge parameter in the with: section
+        if (
+            "auto-merge: 'true'" in existing_content
+            or 'auto-merge: "true"' in existing_content
+        ):
+            print(f"Workflow file {workflow_file} already has auto-merge parameter.")
+            return
+        print(f"Workflow file {workflow_file} missing auto-merge parameter.")
+        print("Updating existing workflow file...")
+
+    workflow_file.write_text(workflow_content, encoding="utf-8")
+    action = "Updated" if file_exists else "Created"
+    print(f"{action} Dependabot auto-merge workflow at {workflow_file}")
+
+
+def disable_codeowners_review_requirement() -> None:  # noqa: C901
```

Comment:
This is nice but it will fail for any non-admin user, right? Not sure if it makes a lot of sense to include it, if it fails gracefully and it is already done, I guess it is fine to keep it.